packages/protocol/lib/test-utils.ts

```diff
@@ -127,7 +127,28 @@ export const assertThrowsAsync = async (promise: any, errorMessage: string = '')
   assert.isTrue(failed, errorMessage)
 }
 
+export async function assertRevertWithReason(promise: any, expectedRevertReason: string = '') {
+  try {
+    await promise
+    assert.fail('Expected transaction to revert')
+  } catch (error) {
+    // When it's a view call, error.message has a shape like:
+    // `Returned error: VM Exception while processing transaction: revert ${revertMessage}`
+    // When it's a transaction (eg a non-view send call), error.message has a shape like:
+    // `Returned error: VM Exception while processing transaction: revert ${revertMessage} -- Reason given: ${revertMessage}.`
+    // Therefore we try to parse the first instance of `${revertMessage}`.
```

Comment:
is this true for `celo-blockchain` and `ganache`? this seems bizarre

Reply:
Only ever tested with ganache, adding a comment indicating such